### PR TITLE
Well list case

### PR DESCRIPTION
**Pull request by taustad**

### CODE REVIEW

File: frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx [hunk 1]
```diff
@@ -48,12 +74,93 @@ type Props = {
...
+                }
+            } catch (error) {
+                console.error()
+            }
+        })()
+    }, [])
+
+    useEffect(() => {
+        if (project.wells) {
+            const a = project.wells.filter((w) => !IsExplorationWell(w))
```
Comment (ahmedoabdi) [inline frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx:107]: Forget to remove unused variables?

File: frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx [hunk 2]
```diff
@@ -48,12 +74,93 @@ type Props = {
...
+        }
+        const wellsDto: Components.Schemas.WellDto[] = wells?.map((w) => Well.toDto(w)) ?? []
+        return wellsDto
+    }
+
+    const saveWellProjectWells = async () => {
+        const newWellProjectWells = wellsToWellsDto(wellProjectWells).filter((w) => w.id === EMPTY_GUID)
+        const updatedWellProjectWells = wellsToWellsDto(wellProjectWells).filter((w) => w.id !== EMPTY_GUID)
+
+        const newWellProjectWellsResult = await (await GetWellService()).createMultipleWells(newWellProjectWells)
```
Comment (ahmedoabdi) [inline frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx:131]: Remove unused variable and set `await GetWellService().createMultipleWells(newWellProjectWells)` ?

File: frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx [hunk 3]
```diff
@@ -48,12 +74,93 @@ type Props = {
...
+        const wellProjectWellsResult = await (await GetWellService()).updateMultipleWells(updatedWellProjectWells)
+        const filteredWellProjectWellsResult = wellProjectWellsResult.filter((w: any) => !IsExplorationWell(w))
+        setWellProjectWells(filteredWellProjectWellsResult)
+    }
+
+    const saveExplorationWells = async () => {
+        const newExplorationWells = wellsToWellsDto(explorationWells).filter((w) => w.id === EMPTY_GUID)
+        const updatedExplorationWells = wellsToWellsDto(explorationWells).filter((w) => w.id !== EMPTY_GUID)
+
+        const newExplorationWellsResult = await (await GetWellService()).createMultipleWells(newExplorationWells)
```
Comment (ahmedoabdi) [inline frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx:142]: Remove unused variable and set `await GetWellService().createMultipleWells(newWellProjectWells)`?

File: frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx [hunk 4]
```diff
@@ -48,12 +74,93 @@ type Props = {
...
+
+        const explorationWellsResult = await (await GetWellService()).updateMultipleWells(updatedExplorationWells)
```
Comment (ahmedoabdi) [inline frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx:144]: Same here another unused variable that can be replaced with service call.

File: frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx [hunk 5]
```diff
@@ -48,12 +74,93 @@ type Props = {
...
+        const filteredExplorationWellsResult = explorationWellsResult.filter((w: any) => IsExplorationWell(w))
+        setExplorationWells(filteredExplorationWellsResult)
+    }
+
+    const handleSave = async () => {
+        // if (!_.isEqual(project.explorationWellCosts, explorationOperationalWellCosts)) {
```
Comment (ahmedoabdi) [inline frontend/src/Components/EditTechnicalInput/EditTechnicalInputModal.tsx:150]: removed commented code if not needed, and refactor variable names to something more descriptive e.g `explorationWellCost` and `developmentwellCost`?

File: frontend/src/Components/EditTechnicalInput/OperationalWellCosts.tsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+    useEffect(() => {
+        if (developmentOperationalWellCosts && setDevelopmentOperationalWellCosts) {
+            const newDevelopmentOperationalWellCosts: DevelopmentOperationalWellCosts = { ...developmentOperationalWellCosts }
+            newDevelopmentOperationalWellCosts.rigUpgrading = developmentRigUpgrading
+            newDevelopmentOperationalWellCosts.rigMobDemob = developmentRigMobDemob
+            newDevelopmentOperationalWellCosts.annualWellInterventionCostPerWell = developmentAnnualWellInterventionCost
+            newDevelopmentOperationalWellCosts.pluggingAndAbandonment = developmentPluggingAndAbandonment
+            setDevelopmentOperationalWellCosts(newDevelopmentOperationalWellCosts)
+            console.log(newDevelopmentOperationalWellCosts)
```
Comment (ahmedoabdi) [inline frontend/src/Components/EditTechnicalInput/OperationalWellCosts.tsx:52]: check rest of console warnings from codacy, and remove if possible.

File: backend/api/Models/Project.cs [hunk 7]
```diff
@@ -20,6 +20,10 @@ public class Project
     public ICollection<Transport>? Transports { get; set; }
     public ProjectPhase ProjectPhase { get; set; }
     public ProjectCategory ProjectCategory { get; set; }
+    public ExplorationOperationalWellCosts? ExplorationOperationalWellCosts { get; set; } // = null!;
+    // public Guid ExplorationOperationalWellCostsId { get; set; }
```
Comment (ahmedoabdi) [inline backend/api/Models/Project.cs:24]: Commented WellCost Guid fields should probably be removed.

File: backend/api/Dtos/ProjectDto.cs [hunk 8]
```diff
@@ -16,6 +16,8 @@ public class ProjectDto
     public DateTimeOffset CreateDate { get; set; }
     public ProjectPhase ProjectPhase { get; set; }
     public ProjectCategory ProjectCategory { get; set; }
+    public ExplorationOperationalWellCostsDto? ExplorationOperationalWellCosts { get; set; } // = null!;
```
Comment (ahmedoabdi) [inline backend/api/Dtos/ProjectDto.cs:19]: commented code.

File: frontend/src/Utils/common.tsx [hunk 9]
```diff
@@ -114,16 +115,20 @@ export const IsInvalidDate = (date?: Date | null): boolean => {
...
-    case 1: name = "Water injection"
-        break
-    case 2: name = "Gas injection"
-        break
-    case 3: name = "WAG"
-        break
-    case 4: name = "Mixed"
-        break
-    default:
-        name = "Depletion"
```
Comment (ahmedoabdi) [inline frontend/src/Utils/common.tsx:126]: format file.

**Review by ahmedoabdi — COMMENTED** [review]
Good job 💯 , check some of the comment. Most of it is just cleaning up, lets discuss more of it on monday 👍🏿

File: frontend/src/Utils/common.tsx [hunk 10]
```diff
@@ -114,16 +115,20 @@ export const IsInvalidDate = (date?: Date | null): boolean => {
...
+        case 4: name = "Mixed"
+            break
+        default:
+            name = "Depletion"
     }
     return name
 }
+
+// eslint-disable-next-line max-len
+// export const IsExplorationWell = (category: Components.Schemas.WellCategory | undefined) => [4, 5, 6].indexOf(category ?? -1) > -1
```
Comment (DanielBohme) [inline frontend/src/Utils/common.tsx:133]: remove commented code

File: backend/api/Services/DevelopmentOperationalWellCostsService.cs [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+            return DevelopmentOperationalWellCostsDtoAdapter.Convert(explorationOperationalWellCosts);
+        }
+        public DevelopmentOperationalWellCosts GetOperationalWellCosts(Guid id)
+        {
+            var operationalWellCosts = _context.DevelopmentOperationalWellCosts!
+                .FirstOrDefault(o => o.ProjectId == id);
+            if (operationalWellCosts == null)
+            {
+                return null!;
+                // throw new ArgumentException(string.Format("OperationalWellCosts {0} not found.", id));
```
Comment (DanielBohme) [inline backend/api/Services/DevelopmentOperationalWellCostsService.cs:50]: is this commented code supposed to be included or removed?

File: backend/api/Services/ExplorationOperationalWellCostsService.cs [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+            return ExplorationOperationalWellCostsDtoAdapter.Convert(explorationOperationalWellCosts);
+        }
+        public ExplorationOperationalWellCosts GetOperationalWellCosts(Guid id)
+        {
+            var operationalWellCosts = _context.ExplorationOperationalWellCosts!
+                .FirstOrDefault(o => o.ProjectId == id);
+            if (operationalWellCosts == null)
+            {
+                return null!;
+                // throw new ArgumentException(string.Format("OperationalWellCosts {0} not found.", id));
```
Comment (DanielBohme) [inline backend/api/Services/ExplorationOperationalWellCostsService.cs:51]: is this commented code supposed to be included or removed?

File: frontend/src/Services/DevelopmentOperationalWellCostsService.ts [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+import { DevelopmentOperationalWellCosts } from "../models/DevelopmentOperationalWellCosts"
+
+export class __DevelopmentOperationalWellCostsService extends __BaseService {
+    public async create(body: Components.Schemas.DevelopmentOperationalWellCostsDto):
+        Promise<Components.Schemas.DevelopmentOperationalWellCostsDto> {
+        const res: Components.Schemas.DevelopmentOperationalWellCostsDto = await this.post("", { body })
+        return DevelopmentOperationalWellCosts.fromJSON(res)
+    }
+
+    public async update(body: Components.Schemas.DevelopmentOperationalWellCostsDto): Promise<Components.Schemas.DevelopmentOperationalWellCostsDto> {
```
Comment (DanielBohme) [inline frontend/src/Services/DevelopmentOperationalWellCostsService.ts:15]: move promise to new line like in line 10 and remove disable max-len

File: frontend/src/Services/ExplorationOperationalWellCostsService.ts [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+import { ExplorationOperationalWellCosts } from "../models/ExplorationOperationalWellCosts"
+
+export class __ExplorationOperationalWellCostsService extends __BaseService {
+    public async create(body: Components.Schemas.ExplorationOperationalWellCostsDto):
+        Promise<Components.Schemas.ExplorationOperationalWellCostsDto> {
+        const res: Components.Schemas.ExplorationOperationalWellCostsDto = await this.post("", { body })
+        return ExplorationOperationalWellCosts.fromJSON(res)
+    }
+
+    public async update(body: Components.Schemas.ExplorationOperationalWellCostsDto): Promise<Components.Schemas.ExplorationOperationalWellCostsDto> {
```
Comment (DanielBohme) [inline frontend/src/Services/ExplorationOperationalWellCostsService.ts:15]: move promise to new line like in line 10 and remove disable max-len

**Review by ahmedoabdi — APPROVED** [review]
LGTM!

**Review by DanielBohme — APPROVED** [review]
LGTM

**Pull request merged**